x-pack/packages/kbn-cloud-security-posture/src/hooks/use_misconfiguration_preview.ts

```diff
@@ -0,0 +1,135 @@
+import type * as estypes from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import {
+  CDR_MISCONFIGURATIONS_INDEX_PATTERN,
+  LATEST_FINDINGS_RETENTION_POLICY,
+  MAX_FINDINGS_TO_LOAD,
+  CspFinding,
+} from '@kbn/cloud-security-posture-common';
+import type { CspBenchmarkRulesStates } from '@kbn/cloud-security-posture-common/schema/rules/latest';
+import { buildMutedRulesFilter } from '@kbn/cloud-security-posture-common';
+import { useKibana } from '@kbn/kibana-react-plugin/public';
+import type { CoreStart } from '@kbn/core/public';
+import { showErrorToast } from '../..';
+import type { CspClientPluginStartDeps } from '../../type';
+import type { FindingsBaseEsQuery } from '../../type';
+import { useGetCspBenchmarkRulesStatesApi } from './use_get_benchmark_rules_state_api';
+
+interface UseFindingsOptions extends FindingsBaseEsQuery {
+  sort: string[][];
+  enabled: boolean;
+  pageSize: number;
+}
+
+type LatestFindingsRequest = IKibanaSearchRequest<estypes.SearchRequest>;
+type LatestFindingsResponse = IKibanaSearchResponse<
+  estypes.SearchResponse<CspFinding, FindingsAggs>
+>;
+
+interface FindingsAggs {
+  count: estypes.AggregationsMultiBucketAggregateBase<estypes.AggregationsStringRareTermsBucketKeys>;
+}
+
+export const getFindingsCountAggQueryMisconfigurationPreview = () => ({
+  count: {
+    filters: {
+      other_bucket_key: 'other_messages',
+      filters: {
+        passed: { match: { 'result.evaluation': 'passed' } },
+        failed: { match: { 'result.evaluation': 'failed' } },
+      },
+    },
+  },
+});
+
+export const getMisconfigurationAggregationCount = (
+  buckets: Array<estypes.AggregationsStringRareTermsBucketKeys | undefined>
+) => {
+  const passed = buckets.find((bucket) => bucket?.key === 'passed');
+  const failed = buckets.find((bucket) => bucket?.key === 'failed');
+  const noStatus = buckets.find((bucket) => bucket?.key === 'other_messages');
+
+  return {
+    passed: passed?.doc_count || 0,
+    failed: failed?.doc_count || 0,
+    no_status: noStatus?.doc_count || 0,
+  };
+};
+
+export const getFindingsQuery = (
+  { query, sort }: UseFindingsOptions,
+  rulesStates: CspBenchmarkRulesStates,
+  pageParam: any
+) => {
+  const mutedRulesFilterQuery = buildMutedRulesFilter(rulesStates);
+
+  return {
+    index: CDR_MISCONFIGURATIONS_INDEX_PATTERN,
+    size: MAX_FINDINGS_TO_LOAD,
+    aggs: getFindingsCountAggQueryMisconfigurationPreview(),
+    ignore_unavailable: false,
+    query: {
+      ...query,
+      bool: {
+        ...query?.bool,
+        filter: [
+          ...(query?.bool?.filter ?? []),
+          {
+            range: {
+              '@timestamp': {
+                gte: `now-${LATEST_FINDINGS_RETENTION_POLICY}`,
+                lte: 'now',
+              },
+            },
+          },
+        ],
+        must_not: [...(query?.bool?.must_not ?? []), ...mutedRulesFilterQuery],
+      },
+    },
+    ...(pageParam ? { from: pageParam } : {}),
+  };
+};
```

Comment:
The `getFindingsQuery` function seems to be doing a lot, I think we can breaking it down into smaller, more manageable functions. This will improve readability and make the code easier to test.

Also I like how we have "builder" functions here, we can rename `getFindingsQuery` to `buildMisconfigurationsFindingsQuery`:

```js
/**
 * Builds the query to fetch the latest findings for misconfigurations findings
 * @param options - The options to build the query
 * @param rulesStates - The rules states to filter the muted rules
 * @param pageParam - The page to fetch
 **/
export const buildMisconfigurationsFindingsQuery = (
  { query }: UseFindingsOptions,
  rulesStates: CspBenchmarkRulesStates,
  pageParam?: number
) => {
  const mutedRulesFilterQuery = buildMutedRulesFilter(rulesStates);

  return {
    index: CDR_MISCONFIGURATIONS_INDEX_PATTERN,
    size: MAX_FINDINGS_TO_LOAD,
    aggs: getFindingsCountAggQueryMisconfigurationPreview(),
    ignore_unavailable: false,
    query: buildMisconfigurationsFindingsQueryWithFilters(query, mutedRulesFilterQuery),
    ...(pageParam ? { from: pageParam } : {}),
  };
};

const buildMisconfigurationsFindingsQueryWithFilters = (
  query: UseFindingsOptions['query'],
  mutedRulesFilterQuery: estypes.QueryDslQueryContainer[]
) => {
  return {
    ...query,
    bool: {
      ...query?.bool,
      filter: [
        ...(query?.bool?.filter ?? []),
        {
          range: {
            '@timestamp': {
              gte: `now-${LATEST_FINDINGS_RETENTION_POLICY}`,
              lte: 'now',
            },
          },
        },
      ],
      must_not: [...(query?.bool?.must_not ?? []), ...mutedRulesFilterQuery],
    },
  };
};
```